### ISSUE TITLE
does not Improves accuracy , returns loss = Nan

### ISSUE THREAD

**Issue by cirograciapons** [wangchuan/MatCaffe-Example#2]
Hi

as the code is configured with cifar , but cifar data is not included, i modified it slightly to run mnist ( lenet) 
in this case code works but it always provides the same output:

epoch: 001, acc_valid: 10.09%, acc_train: 9.91%, loss: NaN
epoch: 002, acc_valid: 10.09%, acc_train: 9.91%, loss: NaN
epoch: 003, acc_valid: 10.09%, acc_train: 9.91%, loss: NaN
